src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -234,6 +234,11 @@ protected function finalizeValue($value)
         }
 
         foreach ($this->children as $name => $child) {
+            if ($child->isDeprecated()) {
+                $msg = sprintf('The child node "%s" at path "%s" is deprecated.', $name, $this->getPath());
```

Comment:
Typically, deprecation message should also mention the version since it's been deprecated, the outcome in the next major version, and the alternative if any. Would be great to allow + encourage doing so here also.